File: src/main/java/Rijndael.java
import javax.crypto.Cipher;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;

public class Rijndael implements IRijndael {
    public byte[] encrypt(byte[] data, byte[] key) {
        //BEGIN_CHALLENGE
        try {
            Cipher aes = Cipher.getInstance("AES/ECB/NOPADDING");
            SecretKey sKey = new SecretKeySpec(key, 0, key.length, "AES");
            aes.init(Cipher.ENCRYPT_MODE, sKey);
            return aes.doFinal(data);
        } catch (Exception ex) {
            ex.toString();
        }
        //END_CHALLENGE
        return null;
    }

    public byte[] decrypt(byte[] data, byte[] key) {
        //BEGIN_CHALLENGE
        try {
            Cipher aes = Cipher.getInstance("AES/ECB/NOPADDING");
            SecretKey sKey = new SecretKeySpec(key, 0, key.length, "AES");
            aes.init(Cipher.DECRYPT_MODE, sKey);
            return aes.doFinal(data);
        } catch (Exception ex) { }
        //END_CHALLENGE    
        return null;
    }
}

File: src/main/scala/Rijndael.scala
//BEGIN_CHALLENGE
import org.bouncycastle.crypto.engines.AESFastEngine
import org.bouncycastle.crypto.params.KeyParameter
//END_CHALLENGE

class Rijndael extends IRijndael {
  override def encrypt(byte: Array[Byte], key: Array[Byte]): Array[Byte] = {
    //BEGIN_CHALLENGE
    transform(byte, key)
    //END_CHALLENGE
  }

  override def decrypt(byte: Array[Byte], key: Array[Byte]): Array[Byte] = {
    //BEGIN_CHALLENGE
    transform(byte, key, encrypt = false)
    //END_CHALLENGE
  }

  //BEGIN_CHALLENGE
  private def transform(byte: Array[Byte], key: Array[Byte], encrypt: Boolean = true): Array[Byte] = {
    val cipher = new AESFastEngine()
    val buffer: Array[Byte] = new Array(byte.length)

    cipher.init(encrypt, new KeyParameter(key))
    cipher.processBlock(byte, 0, buffer, 0)

    buffer
  }
  //END_CHALLENGE
}
